<?xml version="1.0" encoding="utf-8"?>
<!-- EN-Revision: 09c49da6f0167fcdfe53a76e3ea28ecfc0eb337b Maintainer: ae Status: ready --><!-- CREDITS: fabioluciano,ae -->
<refentry xml:id="exception.getprevious" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
 <refnamediv>
  <refname>Exception::getPrevious</refname>
  <refpurpose>Retorna Exception anterior</refpurpose>
 </refnamediv>

 <refsect1 role="description">
  &reftitle.description;
  <methodsynopsis role="Exception">
   <modifier>final</modifier> <modifier>public</modifier> <type class="union"><type>Throwable</type><type>null</type></type><methodname>Exception::getPrevious</methodname>
   <void/>
  </methodsynopsis>
  <para>
   Retorna a Exception anterior (o terceiro valor passado
   no construtor <methodname>Exception::__construct</methodname>).
  </para>
 </refsect1>

 <refsect1 role="parameters">
  &reftitle.parameters;
  &no.function.parameters;
 </refsect1>

 <refsect1 role="returnvalues">
  &reftitle.returnvalues;
  <para>
   Retorna a <classname>Exception</classname> anterior se disponível,
   ou senão &null;.
  </para>
 </refsect1>

 <refsect1 role="examples">
  &reftitle.examples;
  <para>
   <example>
    <title><methodname>Exception::getPrevious</methodname> example</title>
    <para>
     Iterando e imprimindo a pilha de erro.
    </para>
    <programlisting role="php">
<![CDATA[
<?php
class MyCustomException extends Exception {}

function doStuff() {
    try {
        throw new InvalidArgumentException("Primeiro erro lançado.", 112);
    } catch(Exception $e) {
        throw new MyCustomException("Segundo erro lançado.", 911, $e);
    }
}


try {
    doStuff();
} catch(Exception $e) {
    do {
        printf("%s:%d %s (%d) [%s]\n", $e->getFile(), $e->getLine(), $e->getMessage(), $e->getCode(), get_class($e));
    } while($e = $e->getPrevious());
}
?>
]]>
    </programlisting>
    &example.outputs.similar;
    <screen>
<![CDATA[
/home/bjori/ex.php:8 Segundo erro lançado. (911) [MyCustomException]
/home/bjori/ex.php:6 Primeiro erro lançado. (112) [InvalidArgumentException]
]]>
    </screen>
   </example>
  </para>
 </refsect1>

 <refsect1 role="seealso">
  &reftitle.seealso;
  <para>
   <simplelist>
    <member><methodname>Throwable::getPrevious</methodname></member>
   </simplelist>
  </para>
 </refsect1>
</refentry>
<!-- Keep this comment at the end of the file
Local variables:
mode: sgml
sgml-omittag:t
sgml-shorttag:t
sgml-minimize-attributes:nil
sgml-always-quote-attributes:t
sgml-indent-step:1
sgml-indent-data:t
indent-tabs-mode:nil
sgml-parent-document:nil
sgml-default-dtd-file:"~/.phpdoc/manual.ced"
sgml-exposed-tags:nil
sgml-local-catalogs:nil
sgml-local-ecat-files:nil
End:
vim600: syn=xml fen fdm=syntax fdl=2 si
vim: et tw=78 syn=sgml
vi: ts=1 sw=1
-->
